### DISCUSSION TITLE
Transaction control

### DISCUSSION THREAD

**Discussion by fusionbeam** [PostgREST/postgrest#2741] [Q&A] [answered]
Hi,

I have a processing job that may take a few minutes to execute. 
There is a need to provide a status update from within the function, updates required to avoid concurrency conflicts.

I tried to implement it as a PROCEDURE as it allows transaction control and have intermediary commits after status updates.
Calling procedures is not supported in Postgrest currently so I need to ask what would be your advise in this case?
Short of using an external HTTP API for status update I am out of ideas.

Thank you,
Rares

**Comment by steve-chavez** [comment]
Oh, I happen to be working on procedure support. See https://github.com/PostgREST/postgrest/issues/1210#issuecomment-1501220531.

@fusionbeam Could you share your procedure?

**Comment by wolfgangwalther** [comment, marked as answer]
You could also think about implementing this as some kind of tasks queue to which you add with your request. A pg_cron background worker would then actually work on the queue and do the processing - maybe even in batches. pg_cron also allows you to run procedures and you could update the status in the tasks queue table.
